RecoParticleFlow/PFClusterProducer/plugins/alpaka/PFClusterSoAProducerKernel.dev.cc

```diff
@@ -119,13 +121,17 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
       clusterEnergy = seedEnergy;
       tol = pfClusParams.stoppingTolerance();  // stopping tolerance * tolerance scaling
 
-      if (pfRecHits[i].layer() == PFLayer::HCAL_BARREL1)
-        rhENormInv = pfClusParams.recHitEnergyNormInvEB_vec()[pfRecHits[i].depth() - 1];
-      else if (pfRecHits[i].layer() == PFLayer::HCAL_ENDCAP)
-        rhENormInv = pfClusParams.recHitEnergyNormInvEE_vec()[pfRecHits[i].depth() - 1];
-      else {
-        rhENormInv = 0.;
-        printf("Rechit %d has invalid layer %d!\n", i, pfRecHits[i].layer());
+      if (topology.cutsFromDB()) {
+        rhENormInv = (1. / topology[pfRecHits[i].denseId()].noiseThreshold());
```

Comment:
```suggestion
        rhENormInv = (1.f / topology[pfRecHits[i].denseId()].noiseThreshold());
```
should avoid to do the division in double precision.

Can you check if the results are OK ?

Reply:
Validation looks okay when I make this change.